rpc/jsonrpc/server/http_json_handler.go

```diff
@@ -101,7 +103,16 @@ func makeJSONRPCHandler(funcMap map[string]*RPCFunc, logger log.Logger) http.Han
 			logger.Info("HTTPJSONRPC", "method", request.Method, "args", args, "returns", returns)
 			result, err := unreflectResult(returns)
 			if err != nil {
-				responses = append(responses, types.RPCInternalError(request.ID, err))
+				var response types.RPCResponse
+				switch errors.Unwrap(err) {
+				case ctypes.ErrZeroOrNegativeHeight, ctypes.ErrZeroOrNegativePerPage, ctypes.ErrPageOutOfRange:
+					response = types.RPCInvalidRequestError(request.ID, err)
+				case ctypes.ErrHeightExceedsChainHead, ctypes.ErrHeightNotAvailable:
+					response = types.NewRPCSuccessResponse(request.ID, err.Error())
```

Comment:
Can someone check my thinking here. I think requests where the node doesn't have the data in question (i.e. it's lower than the base height or exceeds the nodes height) shouldn't return an error. I was looking into the jsonrpc 2.0 spec and I wouldn't consider this an invalid request. The client hasn't done anything wrong to invalidate it and perhaps the same request to another node would succeed. For HTTP requests the server will just return a `StatusNotFound`. (@melekes, @bez, @marbar3778)

Reply:
I don't think we should return success as it indicates that the call was successfully executed, which is not true. I would return 


32601 | Method not found | The method does not exist / is not available.
-- | -- | --


code along with an error message

Reply:
Hmm but it's not that the method doesn't exist. It's more that the data doesn't exist. Also the call in respect to that it reached the server, was valid and a result was able to be returned would indicate successful execution. I

Reply:
then maybe we should return `result: null`?

Reply:
yeah this may be the closest idiomatic way of saying `StatusNotFound`. But it annoys me because we could include more information that may help the client i.e. the height you requested is above the head of the blockchain or the height you requested is pruned. The point being is that they signal different things. If it's a matter of the node having pruned the block then the client might be able to make the same request to another node. If it's a matter of that the height exceeds the head of the blockchain then the client knows that they must wait before making the request again